### PR TITLE
feat: Sequence sender unit tests

### PR DESCRIPTION
**Pull request by rbpol**

### CODE REVIEW

File: .mockery.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+with-expecter: true
+filename: "{{.InterfaceName | lower }}.generated.go"
```
Comment (rachit77) [inline .mockery.yaml:2]:
```suggestion
filename: "{{.InterfaceName | lower }}_mock_test.go"
```

How about just defining the file name here globally?

File: .mockery.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+mockname: "{{.InterfaceName}}Mock"
+packages:
+  github.com/0xPolygon/cdk/sequencesender:
+    config:
+      dir: ./sequencesender
```
Comment (rachit77) [inline .mockery.yaml:7]: should we place mocks in a separate dir.
Maybe `./sequencesender/mocks`

WDYT?

Comment (rbpol) [reply .mockery.yaml:7]: Cannot do that due to a circle dependencies

File: .mockery.yaml (hunk 1 above)
Comment (rbpol) [reply .mockery.yaml:2]: Cannot do that if we move mocks to "mocks" dir. See below.

File: .mockery.yaml (hunk 2 above)
Comment (rbpol) [reply .mockery.yaml]: Done

File: Makefile [hunk 3]
```diff
@@ -83,6 +90,13 @@ test-seq_sender:
 lint: ## Runs the linter
 	export "GOROOT=$$(go env GOROOT)" && $$(go env GOPATH)/bin/golangci-lint run --timeout 5m
 
+.PHONY: generate
+generate: generate-code-from-proto generate-mocks ## Generates mocks and other autogenerated types
+
+.PHONY: generate-mocks
+generate-mocks: ## Generates mocks
```
Comment (joanestebanr) [inline Makefile:97]: The target `generate-mocks`  is  inside `test/Makefile`. Please, check it. By other hand the naming of the generated mocks doesn't match the rest

Comment (rbpol) [reply Makefile:97]: Fixed

**Comment by rbpol** [comment]
https://github.com/0xPolygon/cdk/pull/103

**Pull request closed without merging**